### PR TITLE
Dynamic routes

### PR DESCRIPTION
**Pull request by cmjosephs**
Dont have screen shots and screen recording due to database outage.

Change route /app to dynamic route /[user-id]
Implement dynamic re routing after login and sign up screens. 
Pass userId props down to NavLayout and SideBar for initial render of profile photo and name.
Set up server side rendering with my-profile and profile menu
Set up client side rendering in my-classes.
Fix bug in AuthProvider that doesn't add mentee to database.

### CODE REVIEW

File: utils/context/AuthProvider.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ export function AuthProvider({ children }) {
   const [user, loading, error] = useAuthState(auth);
   const router = useRouter();
 
-  console.log(loading);
+  console.log(user);
```
Comment (GonzalezEstevan) [inline utils/context/AuthProvider.js]: We could probably delete this if it is not being used anymore

File: utils/context/AuthProvider.js [hunk 2]
```diff
@@ -96,6 +99,7 @@ export function AuthProvider({ children }) {
         var response = await fetch(`/api/users/${user.uid}`)
         var jsonData = await response.json();
         console.log(res)
+        // Change reroute to dynamic route
```
Comment (GonzalezEstevan) [inline utils/context/AuthProvider.js]: I'll work on these once I get this pulled but thank you for the reminders

**Review by GonzalezEstevan — APPROVED** [review]
Top tier coding, there was really only like one console.log but everything else looked clean

File: pages/[user-id]/calendar.js [hunk 3]
```diff
@@ -148,12 +145,18 @@ export default function Calendar() {
   )
 }
 
-// export async function getServerSideProps({ params }) {
+// export async function getServerSideProps(context) {
+//   const userId = context.params['user-id'];
+//   const userInfo = await getUserInfo(userId);
+//   let courseData;
```
Comment (kevinhwkim) [inline pages/[user-id]/calendar.js:151]: 🔥

File: pages/[user-id]/my-courses.js [hunk 4]
```diff
@@ -1,27 +1,21 @@
...
 
 import MyCourses from '../../components/MyCourses/MyCourses.js';
 import MyCoursesExampleData from '../../components/MyCourses/data/MyCourses.example.js';
 import { getUserInfo, getCoursesByMenteeId, getCoursesByMentorId } from '../../utils/api/apiCalls.js';
 
-export default function myCourses() {
+export default function MyCoursesPage() {
+  const router = useRouter();
+  var pathUserId = router.asPath.split('/');
+  const userID = pathUserId[1];
```
Comment (kevinhwkim) [inline pages/[user-id]/my-courses.js:14]: 🙏

File: pages/[user-id]/my-courses.js [hunk 5]
```diff
@@ -55,6 +49,19 @@ export default function myCourses() {
...
+        if (!response?.account_type) {
+          throw new Error('No account type associated with user.')
+        }
+        setUserType(response.account_type)
+      })
+      .catch(err => console.log('Error getting user information: ', err))
+      .then(fetchAllCourses)
+      .catch((err) => console.warn(err))
+  }, []);
+
```
Comment (kevinhwkim) [inline pages/[user-id]/my-courses.js:64]: thanks for refactoring and cleaning this up for me

**Review by kevinhwkim — APPROVED** [review]

File: pages/[user-id]/my-profile.js [hunk 6]
```diff
@@ -9,37 +9,38 @@ import { defaultProfilePic, defaultProfilePicDims } from '../../utils/constants'
...
-    fullName: 'Current User',
-    location: '',
-    description: '',
-    endorsements: 0,
-    photo: ''
+    fullName: `${userInfo.name.first_name} ${userInfo.name.last_name}`,
+    location: userInfo.location,
+    description: userInfo.description,
+    endorsements: userInfo.endorsements,
+    photo: userInfo.photo
```
Comment (tiffanyyv) [inline pages/[user-id]/my-profile.js:21]: Amazing work. Thank you!

File: pages/[user-id]/my-profile.js [hunk 7]
```diff
@@ -62,3 +63,18 @@ export default function MyProfile() {
...
+  if (!userInfo || !userInfo?.account_type) {
+    return {
+      notFound: true,
+    }
+  }
+
+  return {
+    props: { userInfo }
+  }
+}
```
Comment (tiffanyyv) [inline pages/[user-id]/my-profile.js:80]: 🙏🏻🙏🏻

File: utils/customHooks/useFetch.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+        setIsLoading(false);
+      })
+      .catch((err) => {
+        setError(err)
+        setIsLoading(false);
+      })
+  }, [url])
+
+  return { isLoading, error, apiData }
+}
```
Comment (tiffanyyv) [inline utils/customHooks/useFetch.js:23]: Great job handling the loading

**Review by tiffanyyv — COMMENTED** [review]
Amazing job as always

**Pull request merged**